### PR TITLE
[14.0][MIG]Migration of module purchase_comment_template

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

I've migrated the module purchase_incoming_products and its functionalities from Odoo 13.0 to 14.0. I've checked if all unit tests are still running.

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### Notes

I have removed the formated note form the purchase order line, I think this is not required by everyone that needs comments on purchase order. Maybe it can be defined with configuration or I can make a new module to add the format to the lines.

### CODE REVIEW

**Comment by chroi** [comment]
Thanks for your work @alan196!

Can you check why the CI is red and fix it please ?

File: purchase_comment_template/tests/test_purchase_order_report.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        new_purchase.onchange_partner_id_purchase_comment()
+        purchase_dict = new_purchase._convert_to_write(new_purchase._cache)
+        new_purchase = self.env["purchase.order"].create(purchase_dict)
+        self.assertEqual(new_purchase.comment_template2_id, self.after_comment)
+        self.partner_id.property_comment_template_id = self.before_comment.id
+        new_purchase = self.env["purchase.order"].new(vals)
+        new_purchase.onchange_partner_id_purchase_comment()
+        purchase_dict = new_purchase._convert_to_write(new_purchase._cache)
+        new_purchase = self.env["purchase.order"].create(purchase_dict)
+        self.assertEqual(new_purchase.comment_template1_id, self.before_comment)
```
Comment (mmequignon) [inline purchase_comment_template/tests/test_purchase_order_report.py]: Please use `odoo.tests.Form` to test onchange methods

**Comment by sebalix** [comment]
This module should depend on https://github.com/OCA/account-invoice-reporting/tree/14.0/account_comment_template module (it was named `account_invoice_comment_template` before), fix the dependency in the manifest file to get the CI running.

**Comment by leemannd** [comment]
Hello @alan196 I've created a PR with a fix about the dependency. It could be great if you have the time to merge it.
https://github.com/Jarsa-dev/purchase-reporting/pull/1

**Comment by alan196** [comment]
> Hello @alan196 I've created a PR with a fix about the dependency. It could be great if you have the time to merge it.
> [Jarsa-dev#1](https://github.com/Jarsa-dev/purchase-reporting/pull/1)

Thanks for the PR, but I see that there is a module called base_comment_template I think is better to depend on it instead of account_comment_template.

I have updated the PR with the requested changes of tests and change dependency to base_comment_template

**Comment by alan196** [comment]
I see that base_comment_template was changed, I will see the changes and adapt this module.

I'll update later today

**Comment by leemannd** [comment]
Ok. I just saw that there was an huge simplification into the account_comment_template. Maybe there is something to take in account -> https://github.com/OCA/account-invoice-reporting/commit/b6bf4939161c6de8e4aacf35ac25888084f065e3

**Comment by JuanyDForgeflow** [comment]
@alan196 Do you plan to continue this migration?

**Comment by alan196** [comment]
Hello @JuanyDForgeflow I just updated the code and tests to comply how  base_comment_template module works.

Could you help me to review 🙏

**Review by leemannd — APPROVED** [review]
Thanks for the work

**Review by sebalix — APPROVED** [review]

File: purchase_comment_template/readme/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+To install this module, you need to have the module `Invoice Comments
+<https://github.com/OCA/account-invoice-reporting/tree/12.0/account_invoice_comment_template>`_.
```
Comment (AaronHForgeFlow) [inline purchase_comment_template/readme/INSTALL.rst]:
```suggestion
<https://github.com/OCA/account-invoice-reporting/tree/14.0/account_invoice_comment_template>`_.
```

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review LGTM

**Review by JuanyDForgeflow — APPROVED** [review]

**Comment by alan196** [comment]
@AaronHForgeFlow I updated the install section of the readme 👍  I thinks is ready to merge.

I don't know who can help us to merge it. Who do I ping?

**Comment by AaronHForgeFlow** [comment]
@HviorForgeFlow can you have a look? Thanks in advance

**Comment by olaf-wagner** [comment]
We would really like to have this merged, too, so I take the liberty to ping once again, @HviorForgeFlow. Or should we ask somebody else?

**Pull request merged**